enterprise/server/backends/migration_cache/migration_cache.go

```diff
@@ -900,3 +900,7 @@ func (mc *MigrationCache) Stop() error {
 func (mc *MigrationCache) SupportsCompressor(compressor repb.Compressor_Value) bool {
 	return mc.src.SupportsCompressor(compressor) && mc.dest.SupportsCompressor(compressor)
 }
+
+func (mc *MigrationCache) SupportsEncryption(ctx context.Context) bool {
+	return mc.src.SupportsEncryption(ctx) && mc.dest.SupportsEncryption(ctx)
```

Comment:
should we maybe add a precondition check in the migration cache so that we can't accidentally migrate from a cache that supports encryption, to a cache that doesn't?